cmd/preupgradechecks/preupgradechecks.go

```diff
@@ -66,15 +67,33 @@ func makePreUpgradeTaskClient(logger *zap.Logger, fnPodNs, envBuilderNs string)
 // IsFissionReInstall checks if there is at least one fission CRD, i.e. function in this case, on this cluster.
 // We need this to find out if fission had been previously installed on this cluster
 func (client *PreUpgradeTaskClient) IsFissionReInstall() bool {
+	var found bool
 	for i := 0; i < maxRetries; i++ {
+		found = true
 		_, err := client.apiExtClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(FunctionCRD, metav1.GetOptions{})
 		if err != nil && k8serrors.IsNotFound(err) {
-			return false
-
+			found = false
 		}
 	}
 
-	return false
+	return found
+}
+
+// LatestSchemaApplied ensures that the end user has applied the latest CRDs generated to the cluster.
+// For future reference: whenever a new field is added, we need to check for that field's existence in this function
+func (client *PreUpgradeTaskClient) LatestSchemaApplied() error {
+	client.logger.Info("Checking if user has applied the latest CRDs")
+	crd, err := client.apiExtClient.ApiextensionsV1().CustomResourceDefinitions().Get("functions.fission.io", metav1.GetOptions{})
+	if err != nil {
+		client.logger.Error("Could not get the CRD")
+		return err
+	}
+	// Any new field added in Function spec can be checked here provided the substring matches the description in CRD Validation of the field
+	if strings.Contains(crd.Spec.String(), "RequestsPerPod") {
+		return nil
+	}
+	client.logger.Error("Could not find the newer fields")
```

Comment:
Please add field and CRD details.
It could be like
`could not find  Spec.RequestPerPod in functions.fission.io CRD`

Reply:
@sanketsudake Just thinking that we should probably keep this generic as there could be a lot of newer fields and it doesn't make sense to log about each of them . WDYT?

Reply:
I would probably prefer a specific error message to generic error messages.  Having a unique messages for error scenario is more preferred than generic as it makes easier. I have seen even developers going further and assigning a unique identifier to each error messages so that they easily match error logs and source code. This creates predictable behaviour.